server/src/main/java/org/elasticsearch/common/io/stream/BytesCountingFilterInputStream.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.io.stream;
+
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class BytesCountingFilterInputStream extends FilterInputStream {
```

Comment:
nit Seeing there's a `CountingStreamOutput`, maybe it'd be nice if this was renamed `CountingFilterInputStream`.

Reply:
Sounds good. Renamed as suggested.